codebase-pipelines/iam.tf

```
@@ -0,0 +1,164 @@
+      identifiers = ["codebuild.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_role_policy_attachment" "ssm_access" {
+  role       = aws_iam_role.codebase_image_build.name
+  policy_arn = "arn:aws:iam::aws:policy/AmazonSSMReadOnlyAccess"
+}
+
+resource "aws_iam_role_policy" "codebuild_logs" {
+  name   = "log-policy"
+  role   = aws_iam_role.codebase_image_build.name
+  policy = data.aws_iam_policy_document.codebuild_logs.json
+}
+
+data "aws_iam_policy_document" "codebuild_logs" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+      "logs:TagLogGroup"
+    ]
+    resources = [
+      aws_cloudwatch_log_group.codebase_image_build.arn,
+      "${aws_cloudwatch_log_group.codebase_image_build.arn}:*",
+      "arn:aws:logs:${local.account_region}:log-group:*"
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "codebuild:CreateReportGroup",
+      "codebuild:CreateReport",
+      "codebuild:UpdateReport",
+      "codebuild:BatchPutTestCases",
+      "codebuild:BatchPutCodeCoverages"
+    ]
+    resources = [
+      "arn:aws:codebuild:${local.account_region}:report-group/${aws_codebuild_project.codebase_image_build.name}-*",
+      "arn:aws:codebuild:${local.account_region}:report-group/pipeline-${var.application}-*"
+    ]
+  }
+}
+
+resource "aws_iam_role_policy" "ecr_access" {
+  name   = "ecr-policy"
+  role   = aws_iam_role.codebase_image_build.name
+  policy = data.aws_iam_policy_document.ecr_access.json
+}
+
+data "aws_iam_policy_document" "ecr_access" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "ecr:GetAuthorizationToken"
+    ]
+    resources = [
+      "arn:aws:codebuild:${local.account_region}:report-group/pipeline-${var.application}-*"
+    ]
+  }
+
+  statement {
+    # checkov:skip=CKV_AWS_107:GetAuthorizationToken required for ci-image-builder
+    effect = "Allow"
+    actions = [
+      "ecr:GetAuthorizationToken",
+      "ecr-public:GetAuthorizationToken",
+      "sts:GetServiceBearerToken"
+    ]
+    resources = [
+      "*"
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "ecr-public:DescribeImageScanFindings",
+      "ecr-public:GetLifecyclePolicyPreview",
+      "ecr-public:GetDownloadUrlForLayer",
+      "ecr-public:BatchGetImage",
+      "ecr-public:DescribeImages",
+      "ecr-public:ListTagsForResource",
+      "ecr-public:BatchCheckLayerAvailability",
+      "ecr-public:GetLifecyclePolicy",
+      "ecr-public:GetRepositoryPolicy",
+      "ecr-public:PutImage",
+      "ecr-public:InitiateLayerUpload",
+      "ecr-public:UploadLayerPart",
+      "ecr-public:CompleteLayerUpload",
+      "ecr-public:BatchDeleteImage",
+      "ecr-public:DescribeRepositories",
+      "ecr-public:ListImages"
+    ]
+    resources = [
+      "arn:aws:ecr-public::${data.aws_caller_identity.current.account_id}:repository/*"
```

Can the repository be tighter than '*'.

---

Not without a breaking change on the `additional_ecr_repository` property, we currently accept the repo URL and it's not possible to get the ARN from that. We'd have to change it to repo name

---

Might be worth capturing that in a a comment here.
```suggestion
      # We have to wildcard the repository name because we currently expect the repository URL and it's not possible to get the ARN from that
      "arn:aws:ecr-public::${data.aws_caller_identity.current.account_id}:repository/*"
```

---

Comment committed

---

I think I hadn't realised GitHub wasn't showing me all commits. That keeps catching me out.